### ISSUE TITLE
Enhancement: Improve threaded ISR efficiency

### ISSUE THREAD

**Issue by vthakur7f** [Telecominfraproject/OpenCellular#139]
### *Product:* (OC-SDR, OC-LTE, OC-LTE, OC-Power, others)
  OC-SDR 
### *Component:* (PSU, GBC, BBU, FEM, LED, ME)
   GBC/SDR/FE/SYMC/LED
### *Category:* (Hardware, Software/stack, Firmware/bootloader, Documentation)
   Firmware/OCWare
**Describe the solution you'd like**
It will be rare for more than a few interrupts to need to be handled simultaneously anyway, so proposal is changing our library to use a thread pool instead of dedicated threads. From the SYSBIOS documentation, multiple threads can pend on the same semaphore, using a FIFO queue to determine which task wakes up. Extrapolating from here, it should be possible to have multiple tasks pending on the same mailbox.
If this is possible, it should be reasonably easy to spawn a bunch of threads, pend on a mailbox, and then post to the mailbox from our internal ISR

**Is your feature request related to a problem? Please describe.**
As more drivers start implementing threaded interrupts for their alert handling, we need to improve our threaded interrupt library. Right now, we spawn a new task for each interrupt - with 8 temperature and 10 power sensors, this will get out of control very quickly.
